### PR TITLE
Fix `lsan` test suite

### PR DESCRIPTION
**Pull request by sbc100**
After this change the entire `lsan` test suite now passes.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2134,12 +2134,18 @@ def check_memory_setting(setting):
...
-    settings.USE_LSAN = 1
-    default_setting('EXIT_RUNTIME', 1)
-
-    if settings.LINKABLE:
-      exit_with_error('LSan does not support dynamic linking')
+  sanitizer_mem = 0
+  if sanitize:
+    # sanitizers do at least 9 page allocs of a single page during startup.
+    sanitizer_mem += webassembly.WASM_PAGE_SIZE * 9
+    sanitizer_mem += 2097152 * 10
```
Comment (kripken) [inline emcc.py:2141]: please add a comment for this line - I think all the others are explained but this one. that constant in particular looks odd...

Or is this line a dupe of the below? `1 << 21  ===  2097152`, and 10 is almost 11..?

File: emcc.py [hunk 2]
```diff
@@ -2134,12 +2134,18 @@ def check_memory_setting(setting):
...
+    # we also allocate at least 11 "regions". Each region is kRegionSize (2 << 20) but
+    # MmapAlignedOrDieOnFatalError adds another 2 << 20 for alignment.
+    sanitizer_mem += (1 << 21) * 11
+    # When running in the threaded mode asan needs to allocate an array of kMaxNumberOfThreads
+    # (1 << 22) pointers.  See compiler-rt/lib/asan/asan_thread.cpp.
+    if settings.USE_PTHREADS:
+      sanitizer_mem += (1 << 22) * 4
```
Comment (kripken) [inline emcc.py:2148]: This looks like a dupe of new lines 2199-2212.

File: emcc.py [hunk 3]
```diff
@@ -2213,6 +2219,18 @@ def check_memory_setting(setting):
...
+    default_setting('EXIT_RUNTIME', 1)
+
+    if settings.LINKABLE:
+      exit_with_error('LSan does not support dynamic linking')
+
+  if sanitizer_mem:
+    # Increase the size of the initial memory according to how much memory
+    # we think the sanitizers will use.
+    logger.debug(f'adding {sanitizer_mem} bytes of memory for sanitizer support')
+    settings.INITIAL_MEMORY += sanitizer_mem
```
Comment (kripken) [inline emcc.py:2233]: same as above - this looks like a dupe of lines 2199-2212. I found that by searching for `settings.INITIAL_MEMORY += sanitizer_mem` which now appears twice in the code.

**Comment by sbc100** [comment]
OK, this change is a lot simpler now, and IMHO ready to land.

**Review by tlively — APPROVED** [review]

**Pull request merged**